time t1
time t2
---
bool found
time t